catalogue/webapp/pages/search/index.tsx

```diff
@@ -231,15 +223,19 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
         pageSize: 5,
         toggles: serverData.toggles,
       });
-      const works = getQueryResults('works', worksResults);
+      const works = getQueryResults('works', worksResults) as
+        | Work[]
+        | undefined;
 
       // Images
       const imagesResults = await getImages({
         params,
         toggles: serverData.toggles,
         pageSize: 10,
       });
-      const images = getQueryResults('images', imagesResults);
+      const images = getQueryResults('images', imagesResults) as
+        | Image[]
+        | undefined;
```

Comment:
These `as` are fine for now, but when we get Prismic into the catalogue API we should revisit the types on here – we can have TypeScript enforce this for us.

Reply:
Happily, I added these because I was getting warning, but keen to remove them if possible?

Reply:
Ooh, actually I think we can make this work using [TypeScript generics](https://www.typescriptlang.org/docs/handbook/2/generics.html) – I can't remember, have you used those before?

Reply:
Gave it a shot [here](https://github.com/wellcomecollection/wellcomecollection.org/pull/9056/commits/662b0841bfe5d0bdbc08f107f5e9473457babc01)